### PR TITLE
Container fixes & improvements

### PR DESCRIPTION
**Pull request by atenart**

### CODE REVIEW

**Comment by atenart** [comment]
Not planning to add more things to this PR.

File: .github/workflows/build_push_image.yaml [hunk 1]
```diff
@@ -17,10 +20,27 @@ jobs:
...
+              latest_tag=$(
+                  curl -L \
+                      -H "Accept: application/vnd.github+json" \
+                      -H "Authorization: Bearer ${{ github.token }}" \
+                      -H "X-GitHub-Api-Version: 2022-11-28" \
+                      https://api.github.com/repos/${{ github.repository }}/releases/latest |
+                  jq -r '.tag_name'
+              )
+              tags=${{ github.ref_name }}
+              if [ "${latest_tag}" == "${{ github.ref_name }}" ]; then
```
[the remark this answers is not in the document]
Comment (atenart) [reply .github/workflows/build_push_image.yaml:34]: That's intended, `tags` is a list of tags we'll build and in the above example it is currently only containing `github.ref_name`; but I found it more logical to use `github.ref_name` as this is the real value we want to test against which also helps for maintenance (`tags` could contain another tag one day, the conditional could be moved, we can change the tag name for the container, etc).

**Comment by atenart** [comment]
Rebased on top of `main` to resolve a few conflicts and added 2 patches to fix container build with the `python` feature and document the new deps.

**Review by vlrpl — APPROVED** [review]
lgtm, thanks!

**Pull request merged**